### ISSUE TITLE
Immutable.List of length 0 causes UI to freeze

### ISSUE THREAD

**Issue by petermikitsh** [cooperka/react-native-immutable-list-view#16]
Hello, encountered an interesting bug while using `ImmutableVirtualizedList`. Add the following stateless, functional component to your render tree, and will bring your react-native app's UI thread to its knees. With no console error, no nothing.

```jsx
import Immutable from 'immutable';
import ImmutableVirtualizedList from 'react-native-immutable-list-view/lib/ImmutableVirtualizedList';
import React from 'react';

function HomeHelper() {
  const list = new Immutable.List();
  return (
    <ImmutableVirtualizedList
      immutableData={list}
      renderItem={() => (
        <View/>
      )}
    />
  );
}
```

Currently using these dependency versions:

immutable@3.8.1
react-native@0.43.0-rc.3
react-native-immutable-list-view@0.4.0

![ui_broken](https://cloud.githubusercontent.com/assets/1571918/24067159/1e898b52-0b36-11e7-9cb5-5e0c51e2345c.gif)

Using the accessibility inspector to better illustrate the broken UI. Normally the clicks would switch between tabs.

**Comment by petermikitsh** [comment]
I wanted to clarify that this use case will reproduce the issue:

```js
const list = new Immutable.List([]);
```

But this case will **not**:

```js
const list = new Immutable.List([1]);
```

**Comment by cooperka** [comment]
Very weird, I tried this on the example app and got the same problem using RN v0.41.1. Thanks for the bug report! I'll look into it.

**Comment by cooperka** [comment]
@petermikitsh turns out  `getItemCount` was `undefined` for empty lists, which somehow caused the UI to freeze up. It's fixed in `v0.4.1` now.

**Comment by petermikitsh** [comment]
Thanks for the quick turnaround, @cooperka!
